vmd-api/api/openapi.yaml

```diff
@@ -0,0 +1,336 @@
+  license:
+    name: Apache License 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  title: Virtual Machine Manager API of the GHAF project - OpenAPI 3.1
+  version: 1.0.1
+externalDocs:
+  description: Find out more about the GHAF project
+  url: https://github.com/tiiuae/ghaf
+servers:
+- description: The primary server
+  url: https://localhost/api/v1
+- description: The test server
+  url: https://localhost/api/test/v1
+tags:
+- description: Operations of Virtual Machines
+  name: vm
+paths:
+  /vm/list:
+    get:
+      description: Return a IDs list of all Virtual Machines
+      operationId: getVmList
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                items: {}
+          description: IDs list of all Virtual Machines
+        default:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+          description: error payload
+      summary: Get VMs list
+      tags:
+      - vm
+  /vm/info/{id}:
+    get:
+      description: Returns information about the Virtual Machine by ID
+      operationId: getVmInfoById
+      parameters:
+      - description: An ID of the Virtual Machine that needs to be fetched
+        explode: false
+        in: path
+        name: id
+        required: true
+        schema: {}
+        style: simple
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VirtualMachineInfo'
+          description: successful operation
+        "400":
+          description: Invalid Virtual Machine ID supplied
+        "404":
+          description: Virtual Machine ID not found
+        default:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+          description: error payload
+      summary: Finds Virtual Machine info by ID
+      tags:
+      - vm
+  /vm/{action}/{id}:
+    put:
+      description: Make an action to a Virtual Machine by it's ID
+      operationId: vmAction
+      parameters:
+      - description: "Action, which will be applied to a Virtual Machine"
+        example: start
+        explode: false
+        in: path
+        name: action
+        required: true
+        schema:
+          enum:
+          - start
+          - stop
+          - pause
+        style: simple
+      - description: Virtual Machine's ID
+        example: VM#101
+        explode: false
+        in: path
+        name: id
+        required: true
+        schema: {}
+        style: simple
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VirtualMachineInfo'
+          description: successful operation
+        "400":
+          description: Invalid Virtual Machine ID supplied
```

Comment:
Answering my own earlier question - let's define "Invalid" ID as containing:
- any non-numeric characters
- too small (smaller than to be defined)
- too big (larger than to be defined)
@juliuskoskela